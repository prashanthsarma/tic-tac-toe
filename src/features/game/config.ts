export const BoardRows = 3;
export const BoardColumns = 3;
export const MinWinCoinsInStreak = 3;
export const NumOfStreaksToWin = 1;
export const MaxWins = 6;
export const MaxPlayers = 2;

// Used in styles, ideally should use vh or vw to find optimum
export const CoinSize = 147;
export const BoardWidth = CoinSize * BoardColumns;
export const BoardHeight = CoinSize * BoardRows;