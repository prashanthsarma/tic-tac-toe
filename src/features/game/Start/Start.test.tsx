import { describe, it, expect, afterEach } from 'vitest';
import { render, screen, cleanup, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import { Start } from '.';
import { store } from '../../../app/store';
import { MaxPlayers } from '../config';

const renderWithRedux = (component: React.ReactElement) => {
  return {
    ...render(<Provider store={store}>{component}</Provider>),
    store,
  };
};

afterEach(() => {
  cleanup();
  // Reset store state after each test
  store.dispatch({ type: 'game/reset' });
});

describe('Start Component', () => {
  it('should render correctly', () => {
    const { container } = renderWithRedux(<Start />);
    expect(container).toMatchSnapshot();
  });

  it('should have name input for all players', () => {
    renderWithRedux(<Start />);
    const playerInputs = screen.getAllByTestId('player-detail');
    expect(playerInputs).toHaveLength(MaxPlayers);
  });

  it('should update player name in store when input changes', () => {
    const testName = 'First Player';
    renderWithRedux(<Start />);
    
    const playerInputs = screen.getAllByTestId('player-detail-name');
    const firstPlayerInput = playerInputs[0] as HTMLInputElement;
    
    fireEvent.change(firstPlayerInput, { target: { value: testName } });
    
    const state = store.getState();
    expect(state.game.playerStates[1].name).toBe(testName);
  });
});