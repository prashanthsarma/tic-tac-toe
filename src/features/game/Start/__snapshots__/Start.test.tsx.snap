// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="fullPage "
  >
    <div
      class="centrePage"
    >
      <div
        class="outerBorder"
      >
        <div
          class="innerBox"
          style="height: 441px; width: 441px;"
        >
          <form
            class="containerDiv"
          >
            <div
              class="welcomeText"
            >
              <span
                class="textWelcomeTo"
              >
                Welcome to 
              </span>
              <span
                class="textTicTacToe"
              >
                TIC TAC TOE
              </span>
            </div>
            <div
              class="listOfPlayerDetails"
              data-testid="player-detail-list"
            >
              <div
                class="control"
                data-testid="player-detail"
              >
                <label>
                  PLAYER 1
                </label>
                <input
                  data-testid="player-detail-name"
                  maxlength="25"
                  minlength="1"
                  name="name"
                  required=""
                  type="text"
                  value=""
                />
              </div>
              <div
                class="control"
                data-testid="player-detail"
              >
                <label>
                  PLAYER 2
                </label>
                <input
                  data-testid="player-detail-name"
                  maxlength="25"
                  minlength="1"
                  name="name"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
            <button
              class="continueBtn"
            >
              Continue
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
